### ISSUE TITLE
MRMScoring::calcXcorrPrecursorCombinedShapeScore 

### ISSUE THREAD

**Issue by vmusch** [OpenMS/OpenMS#5617]
Should [MRMScoring::calcXcorrPrecursorCombinedShapeScore](https://github.com/OpenMS/OpenMS/blob/e75e5e43ef903f10a263501769fe072f184b89cb/src/openswathalgo/source/OPENSWATHALGO/ALGO/MRMScoring.cpp#L562) really use only the triangular matrix for the calculation? In the initialization of the [xcorr_precursor_combined_matrix_](https://github.com/OpenMS/OpenMS/blob/e75e5e43ef903f10a263501769fe072f184b89cb/src/openswathalgo/source/OPENSWATHALGO/ALGO/MRMScoring.cpp#L213) a complete matrix is created. @hroest

**Comment by jpfeuffer** [comment]
@vmusch @cbielow status?

**Comment by timosachsenberg** [comment]
@singjc can you check this old issue and close if outdated

**Comment by singjc** [comment]
Yeah, I think we can close this. To answer the initial question, only the triangular matrix is used for the calculation most likely to reduce redundant calculations due to symmetry, I assume.

From the `MRMScoring_test` example

### Full matrix

|   | 0       | 1       | 2 | 3       | 4       |
|---|---------|---------|---|---------|---------|
| 0 | 1       | 0.958298| 0 | 0.465706| 0.61671 |
| 1 | 0.958298| 1       | 0 | 0.456762| 0.724542|
| 2 | 0       | 0       | 0 | 0       | 0       |
| 3 | 0.465706| 0.456762| 0 | 1       | 0.396983|
| 4 | 0.61671 | 0.724542| 0 | 0.396983| 1       |

### Triangular matrix

|   | 0       | 1       | 2 | 3       | 4       |
|---|---------|---------|---|---------|---------|
| 0 | 1       | 0.958298| 0 | 0.465706| 0.61671 |
| 1 |         | 1       | 0 | 0.456762| 0.724542|
| 2 |         |         | 0 | 0       | 0       |
| 3 |         |         |   | 1       | 0.396983|
| 4 |         |         |   |         | 1       |
